hazelcast/src/main/java/com/hazelcast/query/impl/getters/ReflectionHelper.java

```diff
@@ -143,31 +147,47 @@ private static Getter createGetter(Object obj, String attribute) {
         try {
             Getter parent = null;
             List<String> possibleMethodNames = new ArrayList<String>(INITIAL_CAPACITY);
-            for (final String name : attribute.split("\\.")) {
+            for (final String fullname : attribute.split("\\.")) {
+                String baseName = removeModifierSuffix(fullname);
+                String modifier = getModifierSuffix(fullname, baseName);
+
                 Getter localGetter = null;
                 possibleMethodNames.clear();
-                possibleMethodNames.add(name);
-                final String camelName = Character.toUpperCase(name.charAt(0)) + name.substring(1);
+                possibleMethodNames.add(baseName);
+                final String camelName = Character.toUpperCase(baseName.charAt(0)) + baseName.substring(1);
                 possibleMethodNames.add("get" + camelName);
                 possibleMethodNames.add("is" + camelName);
-                if (name.equals(THIS_ATTRIBUTE_NAME)) {
-                    localGetter = new ThisGetter(parent, obj);
+                if (baseName.equals(THIS_ATTRIBUTE_NAME.value())) {
+                    localGetter = GetterFactory.newThisGetter(parent, obj);
                 } else {
-                    for (String methodName : possibleMethodNames) {
-                        try {
-                            final Method method = clazz.getMethod(methodName);
-                            method.setAccessible(true);
-                            localGetter = new MethodGetter(parent, method);
-                            clazz = method.getReturnType();
-                            break;
-                        } catch (NoSuchMethodException ignored) {
-                            EmptyStatement.ignore(ignored);
+
+                    if (parent != null) {
+                        clazz = parent.getReturnType();
+                    }
+
+                    if (localGetter == null) {
```

Comment:
no need to have a null check here since `localGetter` is always null, can be fixed after merge